launcher/java/JavaChecker.cpp

```diff
@@ -183,7 +183,7 @@ void JavaChecker::finished(int exitcode, QProcess::ExitStatus status)
     auto os_arch = results["os.arch"];
     auto java_version = results["java.version"];
     auto java_vendor = results["java.vendor"];
-    bool is_64 = os_arch == "x86_64" || os_arch == "amd64" || os_arch == "aarch64" || os_arch == "arm64";
+    bool is_64 = results["sun.arch.data.model"] == "64";
```

Comment:
nope... can't do that
sun isn't available in many jvms that can still run minecraft. you would at minimum need a fallback which kinda defeats the purpose of this PR

Reply:
Do you have an example of a JVM that I can look at? From my understanding, it's just the name that has to do with sun, and not anything to do with the internal implementation package sun (that is indeed not portable).

Reply:
Pretty much and java 6 or 7 jvm (that isn't oracle) to my knowledge.
Sun was an oracle only class and wasn't added to the openjdk until later (during java 8 lifecycle).

Reply:
If we already have to make sure that the guess/backup method works for everything in the first place, then it makes checking the property at all seem redundant and just adds an extra place that needs to be tested. Given the portability issues with the property, I'm starting to think either using a regular expression over `os.arch` or just keeping the hardcoded comparison might be the way to go.

A while ago I experimented with the regular expression idea for another project and came up with `.*64_?([bl]e)?`, which worked with everything I threw at it. We'd still probably want a test on that though.